### DISCUSSION TITLE
Lesson 2: How can you simply pass and address to SimpleStorage without instantiating it?

### DISCUSSION THREAD

**Discussion by tschnide** [smartcontractkit/full-blockchain-solidity-course-py#1827] [Q&A]
Hi, in lesson 2 at about 2:19:00 in the video it goes over this function:

```Solidity
    function sfStore(uint256 _simpleStorageIndex, uint256 _simpleStorageNumber) public {
        SimpleStorage(address(simpleStorageArray[_simpleStorageIndex])).store(_simpleStorageNumber); 
    }
```
I'm curious about what's happening in the body of the function here
`SimpleStorage(address(simpleStorageArray[_simpleStorageIndex])).store(_simpleStorageNumber); ` 

Since it's storing the number using the `.store` function from SimpleStorage, it seems that passing the address to SimpleStorage creates a new instance of SimpleStorage. How can you just call SimpleStorage without the `new` keyword or anything? Can you just pass an address to any contract like this and then interact with the contract stored at that address? Thanks for any help you can provide!

**Comment by cromewar** [comment]
Hello @tschnide sorry for the late reply.
You don't need to use new each time, as you are using this:
`SimpleStorage[] public simpleStorageArray;`

which already declares the array of contract SimpleStorage, and also you are not assigning that instance to a new variable you are assigning the instance to the array.
